### PR TITLE
#2 add options, #19 fix IP parser, update extensions.

### PR DESCRIPTION
**Pull request by Tratcher**
#2, #19, @muratg @blowdart @pakrym 
Clean up the extensions, add an options class.
Bring over the address parser from IISPlatformHandler.
Store the original IP and port in X-Original-For.
Add a sample.

### CODE REVIEW

File: samples/HttpOverridesSample/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNet.Builder;
```
Comment (muratg) [inline samples/HttpOverridesSample/Startup.cs]: License

Comment (Tratcher) [inline samples/HttpOverridesSample/Startup.cs]: Not in samples?

Comment (muratg) [inline samples/HttpOverridesSample/Startup.cs]: :ok:

**Comment by pakrym** [comment]
:ship: :it:

**Pull request merged**